packages/web-components/src/components/va-promo-banner/va-promo-banner.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  h,
+  Prop,
+  State,
+} from '@stencil/core';
+
+const DISMISSED_PROMO_BANNERS_LOCAL_STORAGE_KEY = 'DISMISSED_PROMO_BANNERS';
+@Component({
+  tag: 'va-promo-banner',
+  styleUrl: 'va-promo-banner.css',
+  shadow: true,
+})
+export class VaPromoBanner {
+  @Element() el: HTMLElement;
+
+  /**
+   * The rendering custom markup via the slot instead of the `<a>` with `text` in it
+   * */
+  @Prop() renderCustom: boolean = false;
+
+  /**
+   * `href` attribute for the `<a>` tag. Only gets used if `render` is _not_ used
+   */
+  @Prop() href: string;
+
+  /**
+   * `target` attribute for the `<a>` tag. Only gets used if `render` is _not_ used
+   */
+  @Prop() target: string;
+
+  /**
+   * Controls which icon gets used. Options are 'announcement', 'news', or 'email-signup'.
+   * */
+  @Prop() type: string = 'announcement';
```

Comment:
We could use `'announcement' | 'news' | 'email-signup'` instead of `string` to better represent `type`'s type.

Reply:
Updates have been made